subiquity/models/filesystem.py

```diff
@@ -1588,12 +1588,12 @@ def _remove(self, obj):
     def add_partition(self, device, *, size, offset, flag="", wipe=None,
                       grub_device=None, partition_name=None):
         from subiquity.common.filesystem import boot
-        real_size = align_up(size)
-        log.debug("add_partition: rounded size from %s to %s", size, real_size)
+        #real_size = align_up(size)
+        #log.debug("add_partition: rounded size from %s to %s", size, real_size)
         if device._fs is not None:
             raise Exception("%s is already formatted" % (device,))
         p = Partition(
-            m=self, device=device, size=real_size, flag=flag, wipe=wipe,
+            m=self, device=device, size=size, flag=flag, wipe=wipe,
```

Comment:
While I'm in agreement with this change, I don't think we should batch it with other things and I want to ensure it's tested very well.  I suggest splitting this PR up a bit, particularly with a PR focused on this change and the fallout.

Reply:
oh 100% agreed

Reply:
When you say that you want to see this be very well tested, do you have specific things in mind? I can stare at all the code paths that end up in add_partition and i'm very very nearly convinced there are no problems lurking (possibly there is an issue around the very fine details of resizing a pre-existing partition with an unaligned size, but that's not really very likely to be hit in real life and not actually to do with the proposed change). But it is hard to be absolutely sure!